ibr/demo.js

```diff
@@ -17,13 +17,39 @@ function onChooseFile() {
     const fr = new FileReader();
     fr.onload = function(evt) {
       const bin = evt.target.result;
-      IBRSDK.render( bin,
-          0, document.getElementById('mainCanvas') ); //top level, index 0
+      const out = IBRSDK.render( bin,
+          0, document.getElementById('mainCanvas') ); // top level, index 0
       IBRSDK.createSidebar( bin, document.getElementById('layerList') );
+      document.getElementById('dwn-btn')
+          .addEventListener('click', function(){
+            download( "out.ibr", out );
+          });
     };
     fr.readAsArrayBuffer(file);
   }
 }
 
+/**
+ * Create binary file from data and download the file in browser.
+ * @param {String} filename name of the binary file that will be created.
+ * @param {Buffer} bin binary data that will be saved in the file.
+ */
+function download( filename, bin ) {
```

Comment:
instead of passing in a binary on every render, why not process the data when a user decides download current state?